specs/eip4844/fork-choice.md

```diff
@@ -0,0 +1,105 @@
+# EIP-4844 -- Fork Choice
+
+## Table of contents
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Helpers](#helpers)
+    - [`is_data_available`](#is_data_available)
+- [Updated fork-choice handlers](#updated-fork-choice-handlers)
+  - [`on_block`](#on_block)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the modification of the fork choice according to the executable beacon chain proposal.
+
+## Helpers
+
+#### `is_data_available`
+
+The implementation of `is_data_available` will become more sophisticated during later sharding upgrades.
+Initially, it requires every verifying actor to retrieve the matching `BlobsSidecar`,
+and validate the sidecar with `validate_blobs_sidecar`.
```

Comment:
Do you think we should move `validate_blobs_sidecar` here as well? p2p spec also uses that function but I *think* it's more appropriate here closer to `is_data_available`

On second thought, the dependency graph between the files is a bit strange right now:
* `class BlobsSidecar` defined in p2p spec and p2p spec references `validate_blobs_sidecar()`
* `validate_blobs_sidecar()` defined in beacon chain spec and references `BlobsSideCar`
* `is_data_available()` defined in fork-choice spec and references side ccars and `validate_blobs_sidecar`

I *think* I'd move all of it to the fork-choice and have the fork-choice doc be a clean dependency for both beacon chain and p2p spec

Reply:
> move all of it to the fork-choice and have the fork-choice doc be a clean dependency for both beacon chain and p2p spec

Agreed. It makes more sense according to the dependencies.

Reply:
What about the `class BlobSidecar` def? I think it should go in here as well